File: app/src/main/java/de/bitb/astroskop/ui/main/profiles/ProfilesFragment.java
package de.bitb.astroskop.ui.main.profiles;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.EditText;

import butterknife.BindView;
import butterknife.OnClick;
import de.bitb.astroskop.AstroApplication;
import de.bitb.astroskop.R;
import de.bitb.astroskop.injection.IBind;
import de.bitb.astroskop.model.Profile;
import de.bitb.astroskop.ui.base.NavigationBaseFragment;
import de.bitb.astroskop.ui.base.adapter.SingleRecyclerAdapter;
import de.bitb.astroskop.ui.main.MainActivity;
import de.bitb.astroskop.ui.main.profiles.details.ProfileActivity;

public class ProfilesFragment extends NavigationBaseFragment implements IBind, IProfilesView {

    @BindView(R.id.fragment_profile_list)
    protected RecyclerView recyclerView;

    @BindView(R.id.fragment_profile_edittext_name)
    protected EditText nameET;

    private SingleRecyclerAdapter profileAdapter;

    private ProfilesPresenter presenter;

    public static ProfilesFragment createInstance() {
        return new ProfilesFragment();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        this.presenter = new ProfilesPresenter((AstroApplication) getActivity().getApplication(), this);
        initRecyclerView();
        presenter.onCreate();
    }

    private void initRecyclerView() {
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
        profileAdapter = new ProfilesAdapter(getContext(), presenter);
        recyclerView.setAdapter(profileAdapter);
    }

    @Override
    public void onResume() {
        super.onResume();
        setToolbarTitle("Profiles");
    }

    @Override
    public int getLayoutId() {
        return R.layout.fragment_profiles_overview;
    }

    @Override
    public int getNavigationPosition() {
        return MainActivity.TAB_PROFILES;
    }

    @OnClick(R.id.fragment_profile_button_create)
    protected void onCreateProfile() {
        presenter.createProfile(nameET.getText().toString());
    }

    @Override
    public void refreshView() {
        profileAdapter.notifyDataSetChanged();
    }

    @Override
    public void openProfile(Profile profile) {
        ProfileActivity.startActivity(getContext(), profile.getUuid());
    }

    @Override
    public void openDeleteDialog(Profile profile) {
        dialogBuilder.showDeleteDialog(getContext(), (dialogInterface, i) -> presenter.deleteProfile(profile));
    }

}

File: app/src/main/java/de/bitb/astroskop/ui/main/profiles/ProfilesFragment.kt
package de.bitb.astroskop.ui.main.profiles

import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.EditText

import butterknife.BindView
import butterknife.OnClick
import de.bitb.astroskop.BaseApplication
import de.bitb.astroskop.R
import de.bitb.astroskop.injection.IBind
import de.bitb.astroskop.model.Profile
import de.bitb.astroskop.ui.base.NavigationBaseFragment
import de.bitb.astroskop.ui.base.adapter.SingleRecyclerAdapter
import de.bitb.astroskop.ui.main.MainActivity
import de.bitb.astroskop.ui.main.profiles.details.ProfileActivity

class ProfilesFragment : NavigationBaseFragment(), IBind, IProfilesView {

    @BindView(R.id.fragment_profile_list)
    var recyclerView: RecyclerView? = null

    @BindView(R.id.fragment_profile_edittext_name)
    var nameET: EditText? = null

    private var profileAdapter: SingleRecyclerAdapter<*, *>? = null

    private var presenter: ProfilesPresenter? = null

    override val layoutId: Int
        get() = R.layout.fragment_profiles_overview

    override val navigationPosition: Int
        get() = MainActivity.TAB_PROFILES

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        this.presenter = ProfilesPresenter(activity!!.application as BaseApplication, this)
        initRecyclerView()
        presenter!!.onCreate()
    }

    private fun initRecyclerView() {
        recyclerView!!.layoutManager = LinearLayoutManager(context)
        profileAdapter = ProfilesAdapter(context, presenter)
        recyclerView!!.adapter = profileAdapter
    }

    override fun onResume() {
        super.onResume()
        setToolbarTitle("Profiles")
    }

    @OnClick(R.id.fragment_profile_button_create)
    fun onCreateProfile() {
        presenter!!.createProfile(nameET!!.text.toString())
    }

    override fun refreshView() {
        profileAdapter!!.notifyDataSetChanged()
    }

    override fun openProfile(profile: Profile) {
        ProfileActivity.startActivity(context, profile.uuid)
    }

    override fun openDeleteDialog(profile: Profile) {
        dialogBuilder!!.showDeleteDialog(context) { dialogInterface, i -> presenter!!.deleteProfile(profile) }
    }

    companion object {

        fun createInstance(): ProfilesFragment {
            return ProfilesFragment()
        }
    }

}